gradio/blocks.py

```diff
@@ -429,7 +430,9 @@ async def call_function(self, fn_index, processed_input):
         if inspect.iscoroutinefunction(block_fn.fn):
             prediction = await block_fn.fn(*processed_input)
         else:
-            prediction = await run_in_threadpool(block_fn.fn, *processed_input)
+            prediction = await anyio.to_thread.run_sync(
```

Comment:
running 
`gradio demo/blocks_hello/run.py` 

```
File "F:\SecondaryDownloads\git_repos\gradio\.\demo\blocks_hello\run.py", line 22, in <module>
    demo.launch()
  File "F:\SecondaryDownloads\git_repos\gradio\.\gradio\blocks.py", line 724, in launch
    self.limiter = anyio.run(self.create_limiter, max_threads)
  File "f:\secondarydownloads\git_repos\gradio\venv\lib\site-packages\anyio\_core\_eventloop.py", line 42, in run
    raise RuntimeError(f'Already running {asynclib_name} in this thread')
RuntimeError: Already running asyncio in this thread

```
gives this error, will fix this part. Probably happens because python or uvicorn(probably not uvicorn though)uses asyncio as a default event_loop.

Reply:
Solved this error with synchronize_async function but now I am getting this error, any idea @pcuenca?
```

To create a public link, set `share=True` in `launch()`.
Traceback (most recent call last):
  File "F:\SecondaryDownloads\git_repos\gradio\.\gradio\routes.py", line 283, in predict
    output = await app.blocks.process_api(
  File "F:\SecondaryDownloads\git_repos\gradio\.\gradio\blocks.py", line 521, in process_api
    predictions, duration = await self.call_function(fn_index, processed_input)
  File "F:\SecondaryDownloads\git_repos\gradio\.\gradio\blocks.py", line 435, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "f:\secondarydownloads\git_repos\gradio\venv\lib\site-packages\anyio\to_thread.py", line 28, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(func, *args, cancellable=cancellable,
  File "f:\secondarydownloads\git_repos\gradio\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 793, in run_sync_in_worker_thread
    async with (limiter or current_default_thread_limiter()):
AttributeError: __aenter__

```

Reply:
Oh I see, need to use a callback_function

Reply:
It now works for me after your fixes. Sorry, didn't think about running the demos.